types/ramda/index.d.ts

```diff
@@ -2015,6 +2018,21 @@ declare namespace R {
          */
         nthArg(n: number): (...a: any[]) => any;
 
+        /**
+         * o is a curried composition function that returns a unary function. Like compose, o performs right-to-left function composition.
+         * Unlike compose, the rightmost function passed to o will be invoked with only one argument.
+         * Also, unlike compose, o is limited to accepting only 2 unary functions.
+         * The name o was chosen because of its similarity to the mathematical composition operator ∘.
+         */
+        o<T1, T2, R>(f: (x: T2) => R, g: (x: T1) => T2, v: T1): R;
+        o<T1, T2, R>(f: (x: T2) => R, g: (x: T1) => T2): (v: T1) => R;
+        o<T1, T2, R>(
```

Comment:
I don't think you want to have the type parameter `T1` here because it can't really be inferred at the invocation site.

What about

```ts
o<T2, R>(f: (x: T2) => R): {
    <T1>(g: (x: T1) => T2, v: T1): R;
    <T1>(g: (x: T1) => T2): (v: T1) => R;
}
```

Reply:
I think you are right, I made a mistake.